projects/vdk-plugins/vdk-meta-jobs/src/vdk/plugin/meta_jobs/dag_validator.py

```diff
@@ -141,6 +149,27 @@ def _validate_fail_meta_job_on_error(self, job: Dict):
                 f" is {type(job['fail_meta_job_on_error'])}. Expected type is bool.",
             )
 
+    def _validate_arguments(self, job: Dict):
+        if "arguments" in job:
+            if not isinstance(job["arguments"], dict):
```

Comment:
nit: were the arguments refactored to IJobArguments?

Reply:
Yes, but in the base class SingleJob the arguments are required to be of primitive type (dict in this case) and that's why we expect the arguments to be of type dict here.